### PR TITLE
Add docs for WebXR Depth Sensing - part 2

### PR DESCRIPTION
**Pull request by Elchi3**
Document how to obtain depth data (using the CPU)
Spec: https://immersive-web.github.io/depth-sensing/#obtaining-data

(A third and last PR for depth sensing will deal with the GPU/WebGL depth info.)

### CODE REVIEW

File: files/en-us/web/api/xrcpudepthinformation/data/index.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+- Property
+- Reference
+- VR
+- WebXR
+- WebXR Device API
+browser-compat: api.XRCPUDepthInformation.data
+---
+<div>{{APIRef("WebXR Device API")}}</div>
+
+<p>The <em>read-only</em> <strong><code>data</code></strong> property of the {{DOMxRef("XRCPUDepthInformation")}} interface is an {{jsxref("ArrayBuffer")}} containing depth buffer information in raw format.</p>
```
Comment (sideshowbarker) [inline files/en-us/web/api/xrcpudepthinformation/data/index.html:18]:
```suggestion
<p>The <em>read-only</em> <strong><code>data</code></strong> property of the {{DOMxRef("XRCPUDepthInformation")}} interface is an {{jsxref("ArrayBuffer")}} containing depth-buffer information in raw format.</p>
```

File: files/en-us/web/api/xrcpudepthinformation/data/index.html [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+<p>The data is stored in row-major format, without padding, with each entry corresponding to distance from the view's near plane to the users' environment, in unspecified units. The size of each data entry and the type is determined by {{domxref("XRSession.depthDataFormat", "depthDataFormat")}}. The values can be converted from unspecified units to meters by multiplying them by {{domxref("XRDepthInformation.rawValueToMeters", "rawValueToMeters")}}. The {{domxref("XRDepthInformation.normDepthBufferFromNormView", "normDepthBufferFromNormView")}} property can be used to transform from normalized view coordinates (an origin in the top left corner of the view, with X axis growing to the right, and Y axis growing downward) into depth buffer’s coordinate system.</p>
```
Comment (sideshowbarker) [inline files/en-us/web/api/xrcpudepthinformation/data/index.html:20]:
```suggestion
<p>The data is stored in row-major format, without padding, with each entry corresponding to distance from the view's near plane to the users' environment, in unspecified units. The size of each data entry and the type is determined by {{domxref("XRSession.depthDataFormat", "depthDataFormat")}}. The values can be converted from unspecified units to meters by multiplying them by {{domxref("XRDepthInformation.rawValueToMeters", "rawValueToMeters")}}. The {{domxref("XRDepthInformation.normDepthBufferFromNormView", "normDepthBufferFromNormView")}} property can be used to transform from normalized view coordinates (an origin in the top left corner of the view, with X axis growing to the right, and Y axis growing downward) into the depth buffer’s coordinate system.</p>
```

File: files/en-us/web/api/xrcpudepthinformation/data/index.html [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+<pre class="brush: js">
+const uint16 = new Uint16Array(depthInfo.data);
+const index = column + row * depthInfo.width;
+const depthInMeters = uint16[index] * depthInfo.rawValueToMeters;
+</pre>
+
+<p>(Use {{jsxref("Float32Array")}} for a "float32" data format.)</p>
+
+<p>Note that the depth in meters is in depth buffer coordinates. Additional steps are needed to convert them to normalized view coordinates, or the {{domxref("XRCPUDepthInformation.getDepthInMeters()")}} method can be used.</p>
```
Comment (sideshowbarker) [inline files/en-us/web/api/xrcpudepthinformation/data/index.html:40]:
```suggestion
<p>Note that the depth in meters is in depth-buffer coordinates. Additional steps are needed to convert them to normalized view coordinates, or the {{domxref("XRCPUDepthInformation.getDepthInMeters()")}} method can be used.</p>
```

File: files/en-us/web/api/xrcpudepthinformation/getdepthinmeters/index.html [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+- Method
+- Reference
+- VR
+- WebXR
+- WebXR Device API
+browser-compat: api.XRCPUDepthInformation.getDepthInMeters
+---
+<div>{{APIRef("WebXR Device API")}}</div>
+
+<p>The <strong><code>getDepthInMeters()</code></strong>method of the {{DOMxRef("XRCPUDepthInformation")}} interface returns the depth in meters at (x, y) in normalized view coordinates (origin in the top left corner).</p>
```
Comment (sideshowbarker) [inline files/en-us/web/api/xrcpudepthinformation/getdepthinmeters/index.html:18]:
```suggestion
<p>The <strong><code>getDepthInMeters()</code></strong> method of the {{DOMxRef("XRCPUDepthInformation")}} interface returns the depth in meters at (x, y) in normalized view coordinates (origin in the top left corner).</p>
```

File: files/en-us/web/api/xrcpudepthinformation/getdepthinmeters/index.html [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+<h2>Syntax</h2>
+
+<pre class="brush: js">getDepthInMeters(x, y)</pre>
+
+<h3>Parameters</h3>
+
+<dl>
+  <dt><code>x</code></dt>
+  <dd>X coordinate. (origin at the left, grows to the right).</dd>
```
Comment (sideshowbarker) [inline files/en-us/web/api/xrcpudepthinformation/getdepthinmeters/index.html:28]:
```suggestion
  <dd>X coordinate (origin at the left, grows to the right).</dd>
```

File: files/en-us/web/api/xrcpudepthinformation/index.html [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+{{InheritanceDiagram}}
+
+<p>This interface inherits properties from its parent, {{domxref("XRDepthInformation")}}.</p>
+
+<h2 id="Properties">Properties</h2>
+
+<dl>
+ <dt>{{domxref("XRCPUDepthInformation.data")}} {{ReadOnlyInline}}</dt>
+ <dd>An {{jsxref("ArrayBuffer")}} containing depth buffer information in raw format.</dd>
```
Comment (sideshowbarker) [inline files/en-us/web/api/xrcpudepthinformation/index.html:26]:
```suggestion
 <dd>An {{jsxref("ArrayBuffer")}} containing depth-buffer information in raw format.</dd>
```

File: files/en-us/web/api/xrcpudepthinformation/index.html [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+ <dt>{{domxref("XRDepthInformation.height")}} {{ReadOnlyInline}}</dt>
+ <dd>Contains the height of the depth buffer (number of rows).</dd>
+ <dt>{{domxref("XRDepthInformation.normDepthBufferFromNormView")}} {{ReadOnlyInline}}</dt>
+ <dd>An {{domxref("XRRigidTransform")}} that needs to be applied when indexing into the depth buffer. The transformation that the matrix represents changes the coordinate system from normalized view coordinates to normalized depth buffer coordinates that can then be scaled by depth buffer’s <code>width</code> and <code>height</code> to obtain the absolute depth buffer coordinates.</dd>
```
Comment (sideshowbarker) [inline files/en-us/web/api/xrcpudepthinformation/index.html:30]:
```suggestion
 <dd>An {{domxref("XRRigidTransform")}} that needs to be applied when indexing into the depth buffer. The transformation that the matrix represents changes the coordinate system from normalized view coordinates to normalized depth-buffer coordinates that can then be scaled by depth buffer’s <code>width</code> and <code>height</code> to obtain the absolute depth-buffer coordinates.</dd>
```

File: files/en-us/web/api/xrdepthinformation/height/index.html [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+<p>The <em>read-only</em> <strong><code>height</code></strong> property of the {{DOMxRef("XRDepthInformation")}} interface contains the height of the depth buffer (number of rows).</p>
+
+<h3 id="Value">Value</h3>
+
+<p>An unsigned long integer.</p>
+
+<h2>Examples</h2>
+
+<p>Use {{domxref("XRFrame.getDepthInformation()")}} (CPU) or {{domxref("XRWebGLBinding.getDepthInformation()")}} (WebGL) to obtain depth information. The returned objects will contain the <code>height</code> of the depth buffer which you can use for further calculations.</p>
```
Comment (sideshowbarker) [inline files/en-us/web/api/xrdepthinformation/height/index.html:26]:
```suggestion
<p>Use {{domxref("XRFrame.getDepthInformation()")}} (CPU) or {{domxref("XRWebGLBinding.getDepthInformation()")}} (WebGL) to obtain depth information. The returned objects will contain the <code>height</code> of the depth buffer, which you can use for further calculations.</p>
```

File: files/en-us/web/api/xrdepthinformation/index.html [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  - Reference
+  - WebXR
+  - XR
+  - AR
+  - VR
+browser-compat: api.XRDepthInformation
+---
+<p>{{APIRef("WebXR Device API")}} {{secureContext_header}}</p>
+
+<p>The <code><strong>XRDepthInformation</strong></code> interface contains information about the distance from the user’s device to the real world geometry in the user’s environment.</p>
```
Comment (sideshowbarker) [inline files/en-us/web/api/xrdepthinformation/index.html:16]:
```suggestion
<p>The <code><strong>XRDepthInformation</strong></code> interface contains information about the distance from the user’s device to the real-world geometry in the user’s environment.</p>
```

File: files/en-us/web/api/xrdepthinformation/index.html [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+<p>You will usually interact with these child interfaces. However, <code>XRDepthInformation</code> provides some useful properties that are inherited:</p>
+
+<h2 id="Properties">Properties</h2>
+
+<dl>
+ <dt>{{domxref("XRDepthInformation.height")}} {{ReadOnlyInline}}</dt>
+ <dd>Contains the height of the depth buffer (number of rows).</dd>
+ <dt>{{domxref("XRDepthInformation.normDepthBufferFromNormView")}} {{ReadOnlyInline}}</dt>
+ <dd>An {{domxref("XRRigidTransform")}} that needs to be applied when indexing into the depth buffer. The transformation that the matrix represents changes the coordinate system from normalized view coordinates to normalized depth buffer coordinates that can then be scaled by depth buffer’s <code>width</code> and <code>height</code> to obtain the absolute depth buffer coordinates.</dd>
```
Comment (sideshowbarker) [inline files/en-us/web/api/xrdepthinformation/index.html:35]:
```suggestion
 <dd>An {{domxref("XRRigidTransform")}} that needs to be applied when indexing into the depth buffer. The transformation that the matrix represents changes the coordinate system from normalized view coordinates to normalized depth-buffer coordinates that can then be scaled by depth buffer’s <code>width</code> and <code>height</code> to obtain the absolute depth-buffer coordinates.</dd>
```

File: files/en-us/web/api/xrdepthinformation/normdepthbufferfromnormview/index.html [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+- WebXR Device API
+browser-compat: api.XRDepthInformation.normDepthBufferFromNormView
+---
+<div>{{APIRef("WebXR Device API")}}</div>
+
+<p>The <em>read-only</em> <strong><code>normDepthBufferFromNormView</code></strong> property of the {{DOMxRef("XRDepthInformation")}} interface contains the 3D geometric transform that needs to be applied when indexing into the depth buffer.</p>
+
+<h3 id="Value">Value</h3>
+
+<p>An {{domxref("XRRigidTransform")}} that needs to be applied when indexing into the depth buffer. The transformation that the matrix represents changes the coordinate system from normalized view coordinates to normalized depth buffer coordinates that can then be scaled by depth buffer’s <code>width</code> and <code>height</code> to obtain the absolute depth buffer coordinates.</p>
```
Comment (sideshowbarker) [inline files/en-us/web/api/xrdepthinformation/normdepthbufferfromnormview/index.html:22]:
```suggestion
<p>An {{domxref("XRRigidTransform")}} that needs to be applied when indexing into the depth buffer. The transformation that the matrix represents changes the coordinate system from normalized view coordinates to normalized depth-buffer coordinates that can then be scaled by depth buffer’s <code>width</code> and <code>height</code> to obtain the absolute depth buffer coordinates.</p>
```

File: files/en-us/web/api/xrdepthinformation/normdepthbufferfromnormview/index.html [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+<h2>Examples</h2>
+
+<p>Use {{domxref("XRFrame.getDepthInformation()")}} (CPU) or {{domxref("XRWebGLBinding.getDepthInformation()")}} (WebGL) to obtain depth information. The returned objects will contain the <code>normDepthBufferFromNormView</code> of the depth buffer which you can use for further calculations.</p>
```
Comment (sideshowbarker) [inline files/en-us/web/api/xrdepthinformation/normdepthbufferfromnormview/index.html:26]:
```suggestion
<p>Use {{domxref("XRFrame.getDepthInformation()")}} (CPU) or {{domxref("XRWebGLBinding.getDepthInformation()")}} (WebGL) to obtain depth information. The returned objects will contain the <code>normDepthBufferFromNormView</code> of the depth buffer, which you can use for further calculations.</p>
```

File: files/en-us/web/api/xrdepthinformation/rawvaluetometers/index.html [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+
+<p>For CPU depth information, see also the {{domxref("XRCPUDepthInformation.getDepthInMeters()")}} method.</p>
+
+<h3 id="Value">Value</h3>
+
+<p>A number.</p>
+
+<h2>Examples</h2>
+
+<p>Use {{domxref("XRFrame.getDepthInformation()")}} (CPU) or {{domxref("XRWebGLBinding.getDepthInformation()")}} (WebGL) to obtain depth information. The returned objects will contain the <code>rawValueToMeters</code> scale factor which can be used for further calculations.</p>
```
Comment (sideshowbarker) [inline files/en-us/web/api/xrdepthinformation/rawvaluetometers/index.html:28]:
```suggestion
<p>Use {{domxref("XRFrame.getDepthInformation()")}} (CPU) or {{domxref("XRWebGLBinding.getDepthInformation()")}} (WebGL) to obtain depth information. The returned objects will contain the <code>rawValueToMeters</code> scale factor, which can be used for further calculations.</p>
```

File: files/en-us/web/api/xrdepthinformation/rawvaluetometers/index.html [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+
+<pre class="brush: js">
+const uint16 = new Uint16Array(depthInfo.data);
+const index = column + row * depthInfo.width;
+const depthInMeters = uint16[index] * depthInfo.rawValueToMeters;
+</pre>
+
+<p>(Use {{jsxref("Float32Array")}} for a "float32" data format.)</p>
+
+<p>Note that the depth in meters is in depth buffer coordinates. Additional steps are needed to convert them to normalized view coordinates, or the {{domxref("XRCPUDepthInformation.getDepthInMeters()")}} method can be used.</p>
```
Comment (sideshowbarker) [inline files/en-us/web/api/xrdepthinformation/rawvaluetometers/index.html:40]:
```suggestion
<p>Note that the depth in meters is in depth-buffer coordinates. Additional steps are needed to convert them to normalized view coordinates, or the {{domxref("XRCPUDepthInformation.getDepthInMeters()")}} method can be used.</p>
```

File: files/en-us/web/api/xrdepthinformation/width/index.html [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+
+<p>The <em>read-only</em> <strong><code>width</code></strong> property of the {{DOMxRef("XRDepthInformation")}} interface contains the width of the depth buffer (number of columns).</p>
+
+<h3 id="Value">Value</h3>
+
+<p>An unsigned long integer.</p>
+
+<h2>Examples</h2>
+
+<p>Use {{domxref("XRFrame.getDepthInformation()")}} (CPU) or {{domxref("XRWebGLBinding.getDepthInformation()")}} (WebGL) to obtain depth information. The returned objects will contain the <code>width</code> of the depth buffer which you can use for further calculations.</p>
```
Comment (sideshowbarker) [inline files/en-us/web/api/xrdepthinformation/width/index.html:26]:
```suggestion
<p>Use {{domxref("XRFrame.getDepthInformation()")}} (CPU) or {{domxref("XRWebGLBinding.getDepthInformation()")}} (WebGL) to obtain depth information. The returned objects will contain the <code>width</code> of the depth buffer, which you can use for further calculations.</p>
```

**Review by sideshowbarker — APPROVED** [review]

**Comment by Elchi3** [comment]
The spec uses _depth buffer_ but I'm fine with _depth-buffer_, too.

**Comment by sideshowbarker** [comment]
> The spec uses _depth buffer_ but I'm fine with _depth-buffer_, too.

Yeah, IMHO the spec editors should fix it in the spec too

**Pull request merged**